### PR TITLE
When loading a step based on a run and a job, use the first one

### LINKED ISSUE #2201: Internal Server Error: reload inspector on a step that's been executed twice in one run
Apologies for no audio here: https://www.loom.com/share/bb9757a35ed64d6faa22573f9a12ddb7

Sentry Issue: [LIGHTNING-DA](https://openfn.sentry.io/issues/5462404141/?referrer=github_integration)

```
Ecto.MultipleResultsError: expected at most one result but got 2 in query:

from s0 in Lightning.Invocation.Step,
  join: r1 in assoc(s0, :runs),
  on: r1.id == ^"cb2e9baa-2b6f-4b66-867e-6a33a6f7f244",
  where: s0.job_id == ^"1276bf5f-b7ed-4368-a6ea-0b88ed4ff91b"

  File "lib/ecto/repo/queryable.ex", line 154, in Ecto.Repo.Queryable.one/3
  File "lib/lightning_web/live/workflow_live/edit.ex", line 1426, in LightningWeb.WorkflowLive.Edit.maybe_follow_run/2
  File "lib/lightning_web/live/workflow_live/edit.ex", line 696, in LightningWeb.WorkflowLive.Edit.handle_params/3
  File "lib/phoenix_live_view/utils.ex", line 462, in anonymous fn/5 in Phoenix.LiveView.Utils.call_handle_params!/5
  File "/app/deps/telemetry/src/telemetry.erl", line 321, in :telemetry.span/3
...
(3 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by taylordowns2000**
Fixes #2201 
image

## Validation Steps

- set up a workflow like this
image

- run it
- reload the inspector after switching to inspect the JOB that has multiple steps in this run
- it should pick the first step for that job
- it should not crash

## Review checklist

- [x] I have performed a **self-review** of my code
- [x] I have verified that all appropriate **authorization policies** (`:owner`, `:admin`, `:editor`, `:viewer`) have been implemented and tested
- [x] If needed, I have updated the **changelog**
- [x] Product has **QA'd** this feature

### CODE REVIEW

**Review by elias-ba — APPROVED** [review]
Great fix, thanks @taylordowns2000 !

**Review by christad92 — APPROVED** [review]

**Comment by taylordowns2000** [comment]
Thanks @christad92 and @elias-ba . I'd like to _hold off_ merging this until after @stuartc is done with https://github.com/OpenFn/lightning/pull/2256/files. It will make it easier for him to isolate that change, and then I'll be able to test a new use case for this PR before merging.

So please don't merge for now. I'll wait for Stu, rebase on his thing once it's in main, and then test again and maybe find another edge case that needs addressing.

**Pull request merged**
